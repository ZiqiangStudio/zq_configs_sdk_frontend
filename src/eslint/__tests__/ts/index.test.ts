import { curryExecEslint } from '../util';

describe('TS Lint', () => {
  const execEslint = curryExecEslint('ts');

  test('报错', async () => {
    const { output, error } = await execEslint('error.ts');
    expect(error).toBeTruthy();
    expect(output).toMatchSnapshot();
  });

  test('正常通过', async () => {
    const { output, error } = await execEslint('normal.ts');
    expect(error).toBeFalsy();
    expect(output).toMatchSnapshot();
  });
});
