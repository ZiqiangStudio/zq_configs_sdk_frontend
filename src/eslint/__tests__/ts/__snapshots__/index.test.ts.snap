// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TS Lint 报错 1`] = `" /ts/error.ts    4:10  error  'UsersEntity' is defined but never used                                                      @typescript-eslint/no-unused-vars    8:7   error  'AuthService' is defined but never used                                                      @typescript-eslint/no-unused-vars   14:18  error  Replace \`⏎····name:·string,⏎····password:·string,⏎··\` with \`name:·string,·password:·string\`  prettier/prettier   17:6   error  Unexpected any. Specify a different type                                                     @typescript-eslint/no-explicit-any   18:5   error  Remove this conditional structure or edit its code blocks so that they're not all the same   sonarjs/no-all-duplicated-branches   26:15  error  'password' is assigned a value but never used                                                @typescript-eslint/no-unused-vars   37:9   error  Insert \`,\`                                                                                   prettier/prettier   46:8   error  Insert \`,\`                                                                                   prettier/prettier   47:10  error  Unexpected any. Specify a different type                                                     @typescript-eslint/no-explicit-any  ✖ 9 problems (9 errors, 0 warnings)   3 errors and 0 warnings potentially fixable with the \`--fix\` option.  "`;

exports[`TS Lint 正常通过 1`] = `""`;
